pkg/detector/library/advisory.go

```diff
@@ -4,51 +4,50 @@ import (
 	"fmt"
 	"strings"
 
-	"github.com/Masterminds/semver/v3"
 	"golang.org/x/xerrors"
 
 	"github.com/aquasecurity/trivy-db/pkg/db"
 	dbTypes "github.com/aquasecurity/trivy-db/pkg/types"
-	"github.com/aquasecurity/trivy/pkg/scanner/utils"
+	"github.com/aquasecurity/trivy/pkg/detector/library/comparer"
 	"github.com/aquasecurity/trivy/pkg/types"
 )
 
 // Advisory represents security advisories for each programming language
 type Advisory struct {
-	lang     string
-	comparer comparer
+	ecosystem string
+	comparer  comparer.Comparer
 }
 
 // NewAdvisory is the factory method of Advisory
-func NewAdvisory(lang string) *Advisory {
+func NewAdvisory(ecosystem string, comparer comparer.Comparer) *Advisory {
 	return &Advisory{
-		lang:     lang,
-		comparer: newComparer(lang),
+		ecosystem: ecosystem,
+		comparer:  comparer,
 	}
 }
 
 // DetectVulnerabilities scans buckets with the prefix according to the programming language in "Advisory".
 // If "lang" is python, it looks for buckets with "python::" and gets security advisories from those buckets.
```

Comment:
Should this say "ecosystem" is python?

Reply:
Ah, I forgot to update the comment. I'll replace "python" with "pip".
https://github.com/aquasecurity/trivy-db/pull/86

Reply:
done https://github.com/aquasecurity/trivy/pull/740/commits/b87cca63c08e08891991b19b9d0f4b0e6ab8a420